### ISSUE TITLE
When the TAO config files will be released?

### ISSUE THREAD

**Issue by shwoo93** [SysCV/qdtrack#24]
Hello,

Great work! Thanks for sharing the repo.

I am looking for the TAO results.
When the related codes and config files will be released?

**Comment by OceanPang** [comment]
Hi @shwoo93, we already have a `tao` branch here. However, the configs and trained models are not open public yet. You can refer to this branch for detailed implementations.

I am on another project and will get back to polish all these details later.

**Comment by shwoo93** [comment]
Many thanks. 
I have tried the ```tao``` branch you mentioned.
While I could successfully train and test the model with qdtrack_frcnn_r50_fpn_24e_lvis.py (pre-train stage), I meet the index error when using ft_qdtrack_frcnn_r50_fpn_24e_tao.py (finetune stage).

Specifically, I found the train_ours.json annotation file includes some weird cases: The number of gt_labels and gt_match_indices does not equal (i.e., tracking labels are not fully annotated for every instance)—Obviously, this causes an index error in the following code.

```_match_indices = _gt_match_indices[key_res.pos_assigned_gt_inds]```

Could you please give me any advice?

**Comment by shwoo93** [comment]
As the ```tao``` 'track_id' annotation starts with 0 (which turns out to be the 'instance_id' in COCO-format) the following code in the coco_video_dataset drops it.
```
if ann.get('instance_id', False):
    gt_instance_ids.append(ann['instance_id'])
```

Changing the above code as 
```
instance_id = ann.get('instance_id', None)
if instance_id is not None:
    gt_instance_ids.append(ann['instance_id'])
```
works well.

Will come back to report the final training results.

**Comment by siyliepfl** [comment]
Yes, exactly. Actually, we fixed this bug in the main branch but haven't updated the tao branch yet.

**Comment by shwoo93** [comment]
Thanks for the reply.
I see the training works well.
But the test results show -1 for Track mAP0.5 and mAP0.75. Are there more inconsistencies between the prediction format and gt?

**Comment by siyliepfl** [comment]
> Thanks for the reply.
> I see the training works well.
> But the test results show -1 for Track mAP0.5 and mAP0.75. Are there more inconsistencies between the prediction format and get?

You need to set those thresholds manually. Please refer to the [tao toolkit](uhttps://github.com/TAO-Dataset/taorl). For instance, if you want to have AP75, you need to add that yourself. eg  `tao_eval.params.iou_thrs = [0.75]`

**Comment by shwoo93** [comment]
Training works fine. 
The testing result outputs -1 for AP50 and AP75. Maybe need 

> > Thanks for the reply.
> > I see the training works well.
> > But the test results show -1 for Track mAP0.5 and mAP0.75. Are there more inconsistencies between the prediction format and get?
> 
> You need to set those thresholds manually. Please refer to the tao toolkit. For instance, if you want to have AP75, you can set. eg `tao_eval.params.iou_thrs = [0.75]`

As the tao toolkit processes track_AP50 and AP75 by default (```Line616-618 in tao/toolkit/tao/eval.py```), I believe the reasonable results should come out without a manual setting above.

**Comment by siyliepfl** [comment]
Please check (Line587-591 in tao/toolkit/tao/eval.py)

**Comment by shwoo93** [comment]
By default the params.iou_thrs are set to [0.5], then wouldn't the result of trakc_mAP50 should not be -1? But the results are
![tao0 5](https://user-images.githubusercontent.com/13035722/115075594-459e8880-9f36-11eb-8137-299ef4a1861d.png)

Also, as suggested, I also tried to manually set tao_eval.params.iou_thrs = [0.5, 0.75] before the tao_eval.run(). The results are 
![tao0 75](https://user-images.githubusercontent.com/13035722/115075686-6f57af80-9f36-11eb-9273-698d04c72551.png)
Seems like only operating the last one (i.e., 0.75).

**Comment by siyliepfl** [comment]
Yes, params.iou_thrs is not a list. It should be a numpy array. `np.array([0.5, 0.75])`  Actually, that's a bug of the TAO toolkit.  : )

**Comment by shwoo93** [comment]
Thanks a lot :)
